javapoet/src/main/java/com/palantir/javapoet/TypeSpec.java

```diff
@@ -669,6 +686,51 @@ public Builder addSuperinterface(TypeMirror superinterface, boolean avoidNestedT
             return this;
         }
 
+        public Builder addPermittedSubclasses(Iterable<? extends TypeName> permittedSubclasses) {
```

Comment:
Choose this name to match [`Class.getPermittedSubclasses`](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Class.html#getPermittedSubclasses).

Reply:
Makes sense 👍